src/did/jwt.spec.ts

```diff
@@ -0,0 +1,106 @@
+    privKey = await Secp256k1.generatePrivateKey();
+    const pubKey = Secp256k1.getPublicKeyCompressed(privKey);
+    const did = DidUtil.getDid(pubKey);
+
+    didDocument = {
+      contexts: {
+        values: [ 'https://www.w3.org/ns/did/v1' ],
+      },
+      id: did,
+      controller: undefined,
+      verificationMethods: [
+        {
+          id: `${did}#key1`,
+          type: 'EcdsaSecp256k1VerificationKey2019',
+          controller: did,
+          publicKeyBase58: DidUtil.getPublicKeyBase58(pubKey),
+        },
+      ],
+      authentications: [
+        {
+          verificationMethodId: `${did}#key1`,
+          verificationMethod: undefined,
+        }
+      ],
+      assertionMethods: [
+        {
+          verificationMethodId: undefined,
+          verificationMethod: {
+            id: `${did}#key2`,
+            type: 'EcdsaSecp256k1VerificationKey2019',
+            controller: did,
+            publicKeyBase58: DidUtil.getPublicKeyBase58(pubKey),
+          },
+        }
+      ],
+      keyAgreements: [],
+      capabilityInvocations: [],
+      capabilityDelegations: [],
+      services: [],
+    };
+  });
+
+  it("issue and verify", async () => {
+    const jwt = await DidAuthJwt.issue(didDocument.id, privKey, didDocument.authentications[0].verificationMethodId!, challenge, 3);
+
+    await expect(async () => {
+      await DidAuthJwt.verify(jwt, didDocument, challenge);
+    }).not.toThrow();
+  });
+
+  it("verify: invalid JWT", async () => {
+    await expect(async () => {
+      await DidAuthJwt.verify("dummy jwt", didDocument, "challenge");
+    }).rejects.toThrow();
+  });
+
+  it("verify: unexpected challenge", async () => {
+    const wrongChallenge = "dummy";
+    const jwt = await DidAuthJwt.issue(didDocument.id, privKey, didDocument.authentications[0].verificationMethodId!, wrongChallenge, 3);
+
+    await expect(async () => {
+      await DidAuthJwt.verify(jwt, didDocument, challenge);
+    }).rejects.toThrow(`unexpected challenge: ${wrongChallenge}`);
+  });
+
+  it("verify: expired", async () => {
+    const jwt = await DidAuthJwt.issue(didDocument.id, privKey, didDocument.authentications[0].verificationMethodId!, challenge, 0);
+
+    await sleep(1000);
+
+    await expect(async () => {
+      await DidAuthJwt.verify(jwt, didDocument, challenge);
+    }).rejects.toThrow(`"exp" claim timestamp check failed`);
+  });
+
+  it("verify: signed with wrong private key", async () => {
+    const anotherPrivKey = await Secp256k1.generatePrivateKey();
+    const jwt = await DidAuthJwt.issue(didDocument.id, anotherPrivKey, didDocument.authentications[0].verificationMethodId!, challenge, 3);
+
+    await expect(async () => {
+      await DidAuthJwt.verify(jwt, didDocument, challenge);
+    }).rejects.toThrow();
+  });
+
+  it("verify: wrong verification method ID", async () => {
+    const wrongMethodId = didDocument.assertionMethods[0].verificationMethodId! // use assertionMethods instead of authentications
```

Comment:
`didDocument.assertionMethods[0].verificationMethodId` is undefined, so it might be changed to `didDocument.assertionMethods[0].verificationMethod?.Id` if you intend to test the case of wrong verification method of relationship.

Reply:
Thanks. Good point. That's my mistake. I just fixed it and enhanced comments: c404901c838f9ca18dabe4192658981972740a16